### PR TITLE
Add and fix ECLint in pipeline

### PR DESCRIPTION
**Pull request by drpatelh**
* Add EditorConfig linting similar to https://github.com/nf-core/rnaseq/pull/627
* Simplified default comments everywhere because ECLint was left-aligning them and they looked terrible.
* Changed comment header format in all of the template files too

This took way longer than I wanted it to but the code base should be even nicer and more consistent going forward!! 

All of the tests seem to be happy so I hope I haven't broke something somewhere with all of this re-formatting! Bit late in the day now given we are releasing soon. Oh well 🤷🏽

### CODE REVIEW

File: subworkflows/local/variants_ivar.nf [hunk 1]
```diff
@@ -1,194 +1,194 @@
...
+    //
+    ch_snpeff_vcf      = Channel.empty()
+    ch_snpeff_tbi      = Channel.empty()
+    ch_snpeff_stats    = Channel.empty()
+    ch_snpeff_csv      = Channel.empty()
+    ch_snpeff_txt      = Channel.empty()
+    ch_snpeff_html     = Channel.empty()
+    ch_snpsift_txt     = Channel.empty()
+    ch_snpeff_version  = Channel.empty()
+    ch_snpsift_version = Channel.empty()
```
Comment (JoseEspinosa) [inline subworkflows/local/variants_ivar.nf:111]: Add a line?
```suggestion
    ch_snpsift_version = Channel.empty()
         
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Some minor comments but otherwise awesome as always 👍

File: .markdownlint.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Markdownlint configuration file
```
[the remark this answers is not in the document]
Comment (drpatelh) [reply .markdownlint.yml:1]: These files tend to be in the root of the pipeline repo as standard which is why these have been moved.

File: modules/local/functions.nf [hunk 3]
```diff
@@ -1,19 +1,17 @@
-/*
```
[the remark this answers is not in the document]
Comment (drpatelh) [reply modules/local/functions.nf:1]: Yes, please 👀 Will need to look for these comments in the entire module repo. Copying the `functions.nf` shouldn't be much work. Think I have a script for that 🤔 Problem is that they get left-aligned with the new EClint which makes everything untidy so better to just replace them once and for all everywhere I think.

File: subworkflows/local/variants_ivar.nf (hunk 1 above)
Comment (drpatelh) [reply subworkflows/local/variants_ivar.nf:111]: I tend to not add a line here but if I do it will mean changing stuff everywhere else too 😅

Comment (JoseEspinosa) [reply subworkflows/local/variants_ivar.nf:111]: 🔕 OK, for some reason I thought that you add a line on these cases.

File: .markdownlint.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoseEspinosa) [reply .markdownlint.yml:1]: OK, looks ugly to me, but I have nothing to say if it's the standard 🙄  good to know!

**Comment by drpatelh** [comment]
We good to go here @JoseEspinosa

**Review by JoseEspinosa — APPROVED** [review]
🚀

**Pull request merged**